src/main/java/org/kainos/ea/service/JobRoleService.java

```diff
@@ -27,4 +29,23 @@ public List<JobRole> getAllJobRoles() throws FailedToGetJobRolesException {
         }
     }
 
+    public JobRoleSpec getSpecificationById(int id) throws FailedToGetValidJobId {
+        if (id < 0) {
+            throw new FailedToGetValidJobId();
+        }
+
+        try {
+            JobRoleSpec Spec = jobRoleDao.getSpecificationById(id, databaseConnector.getConnection());
+            if (Spec == null) {
+                throw new FailedToGetValidJobId();
+            }
+            return Spec;
+        } catch (SQLException e) {
+            System.err.println(e.getMessage());
+            throw new FailedToGetValidJobId();
+        } catch (DatabaseConnectionException e) {
+            throw new RuntimeException(e);
```

Comment:
don't throw generic exceptions, you can throw DatabaseConnectionException from this method and catch it in the controller